### ISSUE TITLE
XGboost on Power8 Server

### ISSUE THREAD

**Issue by ghost** [dmlc/xgboost#1661]
For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.
## Environment info

Operating System:
RHEL7
3.10.0-327.33.1.el7.ppc64le #1 SMP Mon Aug 1 05:05:17 EDT 2016 ppc64le ppc64le ppc64le GNU/Linux

Compiler:

Name        : gcc
Arch        : ppc64le
Version     : 4.8.5

Package used (python/R/jvm/C++):

`xgboost` version used:
commit 
Author: Tianqi Chen tqchen@users.noreply.github.com
Date:   Sun Oct 9 20:37:57 2016 -0700

---

If installing from source, please provide 
1. The commit hash (`git rev-parse HEAD`)
   8a7a6dba71dee87d44f0090a256566268766a3b9

If you are using python package, please provide
1. The python version and distribution
2. The command to install `xgboost` if you are not installing from source

If
## Steps to reproduce
1. Remove msse2 CFLAG flag ( I am using an IBM Power64le cpu)
2. Compile xgboost
3. Run Xgboost with config/data that works perfectly on x86_64
4. Xgboost launches but prints no stdout.  Process only uses 1 thread (there are 128 CPU's on this machine). When stracing the process, i see a series of map, munmap calls but no futex.
## What have you tried?
1. Variations of PPC compiler flags here: https://gcc.gnu.org/onlinedocs/gcc/RS_002f6000-and-PowerPC-Options.html
2. Added -Wl,--no-as-needed
   3.

**Comment by phunterlau** [comment]
do you want to try the static link method in #1630 by having a VM for building a static version and copying back to your PPC?

**Comment by pipe2grep** [comment]
@phunterlau so, I figured out my problem.. I was not specifying a model_out in my .conf file.. Once I specified this, xgboost was able to do more than sit there mapping/unmapping memory.   One interesting note is that the binary built by removing unsupported x86 options (-msse2 --fun-roll-loops), the binary will use at most 30-50% of total CPU.  When I compiled using IBM XLC c++ compiler, the binary uses 100% CPU (30-50% USR, the rest SYS).  Big difference is the number of select functions in the latter.
